### PR TITLE
Support Amazon S3 data source in Broker Load 

### PR DESCRIPTION
**Pull request by frwrdt**
issue  #3003

### CODE REVIEW

**Comment by chaoyli** [comment]
Does access s3 need a SDK or dependent library?
This library may be incorporated into third-party

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java [hunk 1]
```diff
@@ -511,7 +598,7 @@ public void pwrite(TBrokerFD fd, long offset, byte[] data) {
             long currentStreamOffset;
             try {
                 currentStreamOffset = fsDataOutputStream.getPos();
-            } catch (IOException e) {
+            } catch (Exception e) {
```
Comment (HangyuanLiu) [inline fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java]: Why catch all exception

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java [hunk 2]
```diff
@@ -307,7 +343,58 @@ public BrokerFileSystem getFileSystem(String path, Map<String, String> propertie
...
+     * file system handle is cached, the identity is host + accessKey_secretKey
+     * @param path
+     * @param properties
+     * @return
+     * @throws URISyntaxException
+     * @throws Exception
+     */
+    public BrokerFileSystem getS3AFileSystem(String path, Map<String, String> properties) {
+        WildcardURI pathUri = new WildcardURI(path);
+        String accessKey = properties.containsKey(FS_S3A_ACCESS_KEY) ? properties.get(FS_S3A_ACCESS_KEY) : "";
```
Comment (HangyuanLiu) [inline fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java]: properties.getOrDefault ?

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java [hunk 3]
```diff
@@ -146,12 +182,12 @@ public BrokerFileSystem getFileSystem(String path, Map<String, String> propertie
         String password = properties.getOrDefault(PASSWORD_KEY, "");
         String dfsNameServices = properties.getOrDefault(DFS_NAMESERVICES_KEY, "");
         String authentication = properties.getOrDefault(CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION,
-            AUTHENTICATION_SIMPLE);
+                AUTHENTICATION_SIMPLE);
```
Comment (HangyuanLiu) [inline fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java]: Please keep indentation

**Review by HangyuanLiu — CHANGES_REQUESTED** [review]

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java (hunk 3 above)
Comment (frwrdt) [reply fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java]: ok

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java (hunk 2 above)
Comment (frwrdt) [reply fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java]: ok

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java (hunk 1 above)
Comment (frwrdt) [reply fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java]: should be 'IOException'

**Comment by frwrdt** [comment]
> Does access s3 need a SDK or dependent library?
> This library may be incorporated into third-party



> Does access s3 need a SDK or dependent library?
> This library may be incorporated into third-party

There is no change in pom.xml and lib for be and fe. 
This pr just add hadoop-aws sdk in pom.xml for broker and there is no conflict in dependencies.

**Review by HangyuanLiu — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**